File: src/test/java/model/TreeParserTest.java
package model;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

public class TreeParserTest {
    @Test
    void testParseInput() {
        var input = Arrays.asList(3,9,20,null,null,15,7);
        var root = TreeParser.parseInput(input);
        assertThat(root.val).isEqualTo(3);
        assertThat(root.left.val).isEqualTo(9);
        assertThat(root.left.left).isNull();
        assertThat(root.left.right).isNull();
        assertThat(root.right.val).isEqualTo(20);
        assertThat(root.right.left.val).isEqualTo(15);
        assertThat(root.right.left.left).isNull();
        assertThat(root.right.left.right).isNull();
        assertThat(root.right.right.val).isEqualTo(7);
        assertThat(root.right.right.left).isNull();
        assertThat(root.right.right.right).isNull();
    }
}

File: python/test/model/tree_parser_test.py
from python.test.model.tree_parser import parse_tree


def test_parse_input():
    data = [3, 9, 20, None, None, 15, 7]
    root = parse_tree(data)
    assert root.val == 3
    assert root.left.val == 9
    assert root.left.left is None
    assert root.left.right is None
    assert root.right.val == 20
    assert root.right.left.val == 15
    assert root.right.left.left is None
    assert root.right.left.right is None
    assert root.right.right.val == 7
    assert root.right.right.left is None
    assert root.right.right.right is None